changebot/blueprints/stale_pull_requests.py

```diff
@@ -86,37 +86,51 @@ def process_pull_requests(repository, installation):
             print('-> PROTECTED by label, skipping')
             yield '-> PROTECTED by label, skipping'
             continue
+
         commit_time = pr.last_commit_date
+        time_since_last_commit = now - commit_time
+
+        # Note: if warning time is before commit time, it's as if the warning
+        # didn't exist since it's no longer relevant.
+        warning_time = pr.last_comment_date('astropy-bot[bot]', filter_keep=is_close_warning)
+        if warning_time is None or warning_time < commit_time:
+            has_warning = False
```

Comment:
Is the `has_warning` flag necessary? It seems like setting `time_since_last_warning = 0` has the same effect anyway, unless you allow `stale_pull_requests_close_seconds` to be negative, which I don't think you do.

Reply:
Good catch!